### PR TITLE
Added templates for gulp + grunt + webpack

### PR DESCRIPTION
**Pull request by NinadKavimandan**
## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [ ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [ ] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [ ] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [ ] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [ ] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Comment by andymckay** [comment]
👋🏾 Thank you for submitting this. As noted in the pull request template and this issue here: https://github.com/actions/starter-workflows/issues/631 we aren't accepting new starter workflows right now. If you believe there should be an exception here, please let me know and we can re-open this pull request.

Sorry about that.

**Comment by NinadKavimandan** [comment]
After discussion with @andymckay and @bishal-pdMSFT, we are going ahead with this contribution. Can you reopen this PR? I don't see the option to do so :(

File: ci/grunt.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Grunt
```
Comment (bishal-pdMSFT) [inline ci/grunt.yml:1]: A bit more descriptive name like "NodeJs with Grunt" might be better.

File: ci/properties/grunt.properties.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Grunt",
+    "description": "Build a Grunt project with npm.",
```
Comment (bishal-pdMSFT) [inline ci/properties/grunt.properties.json:3]: `Build a NodeJS project with npm and grunt`

File: ci/properties/webpack.properties.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Webpack",
+    "description": "Build a Webpack project with npm.",
+    "iconName": "nodejs",
+    "categories": ["JavaScript", "npm", "Webpack"]
```
Comment (bishal-pdMSFT) [inline ci/properties/webpack.properties.json:5]: Shouldn't we change `javascript` to `node`? Should node be added as a language to Linguist?

File: ci/properties/grunt.properties.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    "iconName": "nodejs",
```
Comment (bishal-pdMSFT) [inline ci/properties/grunt.properties.json:4]: Better to have icon for gulp/grunt etc instead of nodejs?

File: ci/webpack.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Build
+      run: |
+        npm install -g webpack webpack-cli --save-dev
```
Comment (bishal-pdMSFT) [inline ci/webpack.yml:27]: can't we assume that webpack-cli is already part of package.json? We assumed that gulp and grunt.

Comment (NinadKavimandan) [reply ci/webpack.yml]: makes sense, taken care of

File: ci/properties/grunt.properties.json (hunk 4 above)
Comment (NinadKavimandan) [reply ci/properties/grunt.properties.json]: yes, updated
This is how these will look on recommending - 
image

File: ci/grunt.yml (hunk 1 above)
Comment (NinadKavimandan) [reply ci/grunt.yml]: agreed. updated

File: ci/properties/grunt.properties.json (hunk 2 above)
Comment (NinadKavimandan) [reply ci/properties/grunt.properties.json]: updated

File: ci/properties/webpack.properties.json (hunk 3 above)
Comment (NinadKavimandan) [reply ci/properties/webpack.properties.json:5]: That would be a redundancy as `scout` already detects `npm` and `yarn`. Language gets higher priority than non-language stacks Keeping `JavaScript` as it is.

File: ci/properties/webpack.properties.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Webpack",
+    "description": "Build a NodeJS project with npm and webpack.",
+    "iconName": "webpack",
+    "categories": ["JavaScript", "npm", "Webpack"]
```
Comment (bishal-pdMSFT) [inline ci/properties/webpack.properties.json:5]: nit: should `npm` be `Npm`?

File: ci/properties/webpack.properties.json (hunk 3 above)
Comment (bishal-pdMSFT) [reply ci/properties/webpack.properties.json:5]: Existing node [template](https://github.com/actions/starter-workflows/blob/main/ci/properties/node.js.properties.json) specifies both "Javascript" and "node". How should we rationalize these two?
In other words, javascript is the language and node is the runtime. Should we treat them differently or as you say npm/yarn is a proxy for node runtime?

Comment (NinadKavimandan) [reply ci/properties/webpack.properties.json:5]: "Node" in the existing template really has no mapping with any property of the repo. Node can be consumed in multiple ways based on the packagemanager, and the workflow changes as per the packagamanager used. Thus I feel proxying node runtime with npm/yarn should be the way to go

File: ci/properties/webpack.properties.json (hunk 6 above)
Comment (NinadKavimandan) [reply ci/properties/webpack.properties.json:5]: `npm` should be fine, commercially it is known as `npm` only.

File: ci/properties/webpack.properties.json (hunk 3 above)
Comment (NinadKavimandan) [reply ci/properties/webpack.properties.json]: Also adding "TyepScript" as a dependency, as these all can be used with ts - https://www.typescriptlang.org/docs/handbook/integrating-with-build-tools.html

**Review by ashwinsangem — APPROVED** [review]
LGTM

**Comment by theyemeni** [comment]
says outdated?

**Comment by NinadKavimandan** [comment]
@theyemeni didn't get your concern?

**Pull request merged**